packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -231,6 +231,9 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
                 logCounter: LogUtils.getCounter(),
               })
             } catch (err: any) {
+              // Release the request if 'run:origin:fn' fails
+              Cypress.backend('ready:for:origin', { originPolicy: location.originPolicy })
```

Comment:
I wonder if we should consider this a failure scenario and pass `failed: true`? If so, we'd need to conditionally call `cross:origin:finished`.

Reply:
That seems more correct. I was missing that you shouldn't call `cross:origin:finished` before